lib/internal/bootstrap_node.js

```diff
@@ -361,20 +361,19 @@
         }
 
       } else {
-        // If we handled an error, then make sure any ticks get processed
-        NativeModule.require('timers').setImmediate(process._tickCallback);
-
         // Emit the after() hooks now that the exception has been handled.
         if (async_hook_fields[kAfter] > 0) {
           do {
             NativeModule.require('async_hooks').emitAfter(
               async_uid_fields[kCurrentAsyncId]);
-          // popAsyncIds() returns true if there are more ids on the stack.
-          } while (popAsyncIds(async_uid_fields[kCurrentAsyncId]));
+          } while (asyncIdStackSize() > 0);
```

Comment:
Few questions:
1. why is this better? feels less abstract/more "internal implementation exposed"
2. who actually pops the ids?
3. cache `emitAfter`?
```js
  const emitAfter = NativeModule.require('async_hooks').emitAfter;
  do {
    emitAfter(async_uid_fields[kCurrentAsyncId]);
  }...
```